/* Copyright 2025 The OpenXLA Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package xla.gpu;

import "xla/service/gpu/backend_configs.proto";
import "xla/xla_data.proto";

message GpuNormDescriptorProto {
  xla.gpu.CudnnNormBackendConfig backend_config = 1;
  xla.ShapeProto x_shape = 2;
  xla.ShapeProto scale_shape = 3;
  optional xla.ShapeProto bias_shape = 4;
  xla.ShapeProto y_or_dx_shape = 5;
  optional xla.ShapeProto expectation_shape = 6;
  optional xla.ShapeProto norm_factor_shape = 7;
  optional xla.ShapeProto dy_shape = 8;
  optional xla.ShapeProto dscale_shape = 9;
  optional xla.ShapeProto dbias_shape = 10;
  int64 scratch_size = 11;
}
